inst/include/RcppSimdJson/deserialize.hpp

```diff
@@ -422,12 +422,12 @@ inline auto deserialize(simdjson::dom::element parsed, const Parse_Opts& parse_o
 
 
 template <typename json_T, bool is_file>
-inline simdjson::simdjson_result<simdjson::dom::element> parse(simdjson::dom::parser& parser,
+inline simdjson::simdjson_result<simdjson::ondemand::document> parse(simdjson::ondemand::parser& parser,
                                                                const json_T&          json) {
     if constexpr (utils::resembles_vec_raw<json_T>()) {
         /* if `json` is a raw (unsigned char) vector, we can cheat */
-        return parser.parse(
-            std::string_view(reinterpret_cast<const char*>(&(json[0])), std::size(json)));
+        simdjson::padded_string content =simdjson::padded_string(std::string_view(reinterpret_cast<const char*>(&(json[0])), std::size(json)));
```

Comment:
This is unsafe because the padding string will go out of scope (and be destroyed) at the return of the function, so you will have a document, but not its input string... so the document will be left pointing at data that might get reclaimed or overwritten at any time.